engine/consensus/dkg/reactor_engine.go

```diff
@@ -158,6 +159,34 @@ func (e *ReactorEngine) EpochSetupPhaseStarted(currentEpochCounter uint64, first
 	e.registerPhaseTransition(curDKGInfo.phase3FinalView, dkgmodule.Phase3, e.end(nextEpochCounter))
 }
 
+// EpochCommittedPhaseStarted handles the EpochCommittedPhaseStarted protocol event. It
+// compares the key vector locally produced by Consensus nodes against the FlowDKG smart contract
+// key vectors. If the keys don't match a log statement will be invoked. In the happy case the locally
+// produced key should match, if the keys do not match the node will have a invalid random beacon key.
+func (e *ReactorEngine) EpochCommittedPhaseStarted(currentEpochCounter uint64, first *flow.Header) {
+	nextDKG, err := e.State.Final().Epochs().Next().DKG()
+	if err != nil {
+		e.log.Fatal().Err(err).Msg("could not retrieve next DKG info")
```

Comment:
A few comments here:
* I would include something in the message that makes it obvious this error occurred within the scope of this sanity check, like `checking DKG key consistency: could not retrieve next DKG info`
* I think it's OK to use `Error` log level for these. In particular, the case where the node failed to complete the DKG locally (though globally the DKG succeeded) is a case we need to handle in the software and shouldn't cause a crash